src/Razor/benchmarks/Microsoft.AspNetCore.Razor.Performance/TagHelperResolutionResultSerializationBenchmark.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Performance
+{
+    public class TagHelperResolutionResultSerializationBenchmark
+    {
+        private readonly byte[] _tagHelperBuffer;
+
+        public TagHelperResolutionResultSerializationBenchmark()
+        {
+            var current = new DirectoryInfo(AppContext.BaseDirectory);
+            while (current != null && !File.Exists(Path.Combine(current.FullName, "taghelpers.json")))
+            {
+                current = current.Parent;
+            }
+
+            var tagHelperFilePath = Path.Combine(current.FullName, "taghelpers.json");
+            _tagHelperBuffer = File.ReadAllBytes(tagHelperFilePath);
+
+            // Deserialize from json file.
+            OriginalSerializer = new JsonSerializer();
+            OriginalSerializer.Converters.Add(new TagHelperDescriptorJsonConverter());
+
+            EnhancedSerializer = new JsonSerializer();
+            EnhancedSerializer.Converters.Add(new TagHelperDescriptorJsonConverter());
+            EnhancedSerializer.Converters.Add(new TagHelperResolutionResultJsonConverter());
+            using (var stream = new MemoryStream(_tagHelperBuffer))
+            using (var reader = new JsonTextReader(new StreamReader(stream)))
+            {
+                var tagHelpers = EnhancedSerializer.Deserialize<IReadOnlyList<TagHelperDescriptor>>(reader);
+                TagHelperResolutionResult = new TagHelperResolutionResult(tagHelpers, Array.Empty<RazorDiagnostic>());
+            }
+        }
+
+        public JsonSerializer OriginalSerializer { get; set; }
+        public JsonSerializer EnhancedSerializer { get; set; }
+        private TagHelperResolutionResult TagHelperResolutionResult { get; }
+
+        [Benchmark(Description = "Razor TagHelperResolutionResult Roundtrip JObject Serialization")]
+        public void TagHelper_JObject_Serialization_RoundTrip()
```

Comment:
It's good to capture this once to ensure we've improved the perf benchmark but I wouldn't keep it around. The other method represents how we want to deserialize everything though so definitely keep that around.

Reply:
Removed!